cmd/query/app/static_handler_test.go

```diff
@@ -266,8 +265,10 @@ func TestHotReloadUIConfigTempFile(t *testing.T) {
 	c := string(h.indexHTML.Load().([]byte))
 	assert.Contains(t, c, "About Jaeger")
 
-	newContent := strings.Replace(string(content), "About Jaeger", "About a new Jaeger", 1)
-	err = syncWrite(tmpFileName, []byte(newContent), 0644)
+	newContent, err := os.ReadFile("fixture/ui-config-new-hotreload.json")
```

Comment:
For my learning, how does reading a file's contents instead of performing a string replace fix this flaky test?

Reply:
Yeah, I suspect the issue is that we're writing to the same file, sync or not, and the reader happens to read it while it's empty. A safer way is to write to another file and do a move into the watched file name.